.circleci/wait-for-job-finish.sh

```diff
@@ -6,24 +6,32 @@ set -o nounset
 # Use the error status of the first failure, rather than that of the last item in a pipeline.
 set -o pipefail
 
+trap 's=$?; echo -e >&2 "\nError in $0:\nat line "$LINENO": $BASH_COMMAND"; exit $s' ERR
+
+date
 counter=0
 
 # Wait up to 25 minutes for job to finish. A job can run on multiple nodes: parallelism > 1
 while [ "$counter" -le 1500 ]; do
   # Find number of nodes in running
   job_detail=$(curl -s "https://circleci.com/api/v2/project/github/DataBiosphere/terra-ui/job/$CIRCLE_BUILD_NUM")
-  job_running_nodes_count=$(echo "$job_detail" | jq -r '.parallel_runs[] | select(.status == "running") | select(.index != '"$CIRCLE_NODE_INDEX"')' | grep -c "running")
 
-  if [ "$job_running_nodes_count" -eq 0 ]; then
-    exit 0
+  # First, check the response for valid job json
+  runs=$(echo "$job_detail" | jq -r '.parallel_runs[]')
+  running_vm=$(echo "$runs" | jq -r 'select(.status=="running")')
+  count=$(echo "$running_vm" | grep -c -e "running" || test $? = 1;)
+  if [ "$count" -eq 0 ]; then
+      echo "Parallel runs have finished. Exiting wait-for-job-finish script."
+      echo "$runs"
+      exit 0
```

Comment:
Additional logs for debugging or manual inspection.

<img width="817" alt="Screenshot 2024-05-15 at 9 40 38 PM" src="https://github.com/DataBiosphere/terra-ui/assets/35533885/88593991-642c-4fe6-9ddc-38304c966d4e">